diff -Naur linux-4.4/sound/soc/tegra/tegra_max98090.c linux-4.4.new/sound/soc/tegra/tegra_max98090.c
--- a/sound/soc/tegra/tegra_max98090.c	2016-01-10 16:01:32.000000000 -0700
+++ b/sound/soc/tegra/tegra_max98090.c	2016-01-20 17:29:31.202294069 -0700
@@ -110,7 +110,7 @@
 	.name = "Headphone detection",
 	.report = SND_JACK_HEADPHONE,
 	.debounce_time = 150,
-	.invert = 1,
+	.invert = 0,
 };
 
 static struct snd_soc_jack tegra_max98090_mic_jack;
